typescript/infra/src/config/agent.ts

```diff
@@ -74,6 +74,8 @@ interface ValidatorSet {
 interface Validator {
   address: string;
   checkpointSyncer: CheckpointSyncerConfig;
+  identifier?: string;
+  readonly?: boolean;
```

Comment:
does identifier being defined imply readonly? maybe we can remove this

Reply:
Seems like the existence of one implies the other, could maybe consider something along the lines of

```
type Validator = {
  address: string;
  checkpointSyncer: CheckpointSyncerConfig;
  readonly: false,
} | {
  address: string;
  checkpointSyncer: CheckpointSyncerConfig;
  readonly: false,
  identifier: string;
}
```